web/src/stats/stats.ts

```diff
@@ -394,34 +463,48 @@ function populate_messages_sent_over_time(data) {
     // graph to any bar graph, since otherwise the rangeselector shows both (plotly bug)
     let clicked_cumulative = false;
 
-    function draw_or_update_plot(rangeselector, traces, last_value_is_partial, initial_draw) {
+    function draw_or_update_plot(
+        rangeselector: Partial<Plotly.RangeSelector>,
+        traces: DataByUserType<Partial<Plotly.PlotData>>,
+        last_value_is_partial: boolean,
+        initial_draw: boolean,
+    ): void {
         $("#daily_button, #weekly_button, #cumulative_button").removeClass("selected");
         $("#id_messages_sent_over_time > div").removeClass("spinner");
         if (initial_draw) {
             traces.human.visible = true;
             traces.bot.visible = "legendonly";
             traces.me.visible = "legendonly";
         } else {
-            const plotDiv = document.querySelector("#id_messages_sent_over_time");
-            traces.me.visible = plotDiv.data[0].visible;
-            traces.human.visible = plotDiv.data[1].visible;
-            traces.bot.visible = plotDiv.data[2].visible;
+            const plotDiv = document.querySelector<Plotly.PlotlyHTMLElement>(
+                "#id_messages_sent_over_time",
+            )!;
+            traces.me.visible = (plotDiv.data[0] as Plotly.PlotData).visible;
```

Comment:
Do we still need this after the DefinitelyTyped change?

Reply:
Not resolved by upgrading DT.  `plotDiv.data[0]` is a `Plotly.Data` type infered by TS compiler. However, from `.visible` field I infer that `plotDiv.data[0]` should be a narrower type, `Plotly.PlotData`, a children class of `Plotly.Data`.  So type casting is still needed at this point.